course_discovery/apps/edly/api/v1/tests/test_views.py

```diff
@@ -0,0 +1,81 @@
+"""
+Unit tests for Edly API views.
+"""
+from rest_framework.reverse import reverse
+from course_discovery.apps.api.v1.tests.test_views.mixins import (
+    APITestCase,
+    SerializationMixin
+)
+from course_discovery.apps.core.tests.factories import (
+    USER_PASSWORD,
+    UserFactory
+)
+from course_discovery.apps.course_metadata.tests.factories import (
+    CourseRunFactory,
+    OrganizationFactory
+)
+from course_discovery.apps.course_metadata.models import (
+    Program
+)
+
+
+class ProgramViewSetTests(SerializationMixin, APITestCase):
+    program_list_url = reverse('edly:v1:program-list')
+
+    def setUp(self):
+        super(ProgramViewSetTests, self).setUp()
+        self.user = UserFactory(is_staff=True, is_superuser=True)
+        self.client.login(username=self.user.username, password=USER_PASSWORD)
+
+    def test_create_with_authentication(self):
+        """
+        Verify endpoint successfully creates a program.
+        """
+        response = self.client.post(self.program_list_url, self._program_data(), format='json')
+        program = Program.objects.last()
+        assert response.status_code == 201
+        assert program.title == response.data['title']
+        assert program.status == response.data['status']
+
+    def test_update_with_authentication(self):
+        """
+        Verify endpoint successfully updates a program.
+        """
+        program_data = self._program_data()
+
+        response = self.client.post(self.program_list_url, self._program_data(), format='json')
+
+        program = Program.objects.last()
+
+        program_detail_url = reverse('edly:v1:program-detail', kwargs={'pk': program.id})
+        program_data['title'] = '{orignal_title} Test Update'.format(orignal_title=program_data['title'])
+        program_data['status'] = 'unpublished'
+        response = self.client.put(program_detail_url, program_data, format='json')
+        program = Program.objects.last()
+        assert response.status_code == 200
+        assert program.title == response.data['title']
+        assert program.status == response.data['status']
+
+    def test_create_without_authentication(self):
+        """
+        Verify authentication is required when creating a program.
+        """
+        self.client.logout()
+        Program.objects.all().delete()
+
+        response = self.client.post(self.program_list_url)
+        assert response.status_code == 403
+        assert Program.objects.count() == 0
+
+    def _program_data(self):
```

Comment:
private methods usually come before test methods to make it easier to read in order.